File: Example/Sources/TestMethod.cs
using System;
using System.Collections.Generic;

namespace DataTransferObjects
{
  /// <summary>
  /// メソッドテストクラス
  /// </summary>
  public class TestMethod
  {
    public class InnerClass
    {
      public static string StaticField = "789";

      public static string StaticMethod()
      {
        return "bbb";
      }

      public static string StaticMethodArg(string name)
      {
        return "hey!" + name;
      }
    }
    public static string StaticMethod()
    {
      return "ccc";
    }

    /// <summary>
    /// 定数フィールド
    /// </summary>
    public const string ConstField = "123";

    /// <summary>
    /// プロパティ例
    /// </summary>
    public string prop { set; get; } = "";

    public int propInt { set; get; } = 50;

    /// <summary>
    /// メソッド例
    /// </summary>
    public void Method()
    {
      // this.prop = TestMethod.StaticMethod()
      prop = StaticMethod();
      // this.prop = TestLogic.StaticMethod()
      prop = TestMethod.StaticMethod();

      // this.prop = TestMethod.InnerClass.StaticMethod()
      prop = InnerClass.StaticMethod();
      // this.prop = TestMethod.InnerClass.StaticMethod()
      prop = TestMethod.InnerClass.StaticMethod();
      // this.prop = TestMethod.InnerClass.StaticMethodArg("test")
      prop = TestMethod.InnerClass.StaticMethodArg("test");
      // this.prop = TestMethod.InnerClass.StaticMethodArg(propInt.toString())
      prop = TestMethod.InnerClass.StaticMethodArg(propInt.ToString());

      // this.prop = OtherClass.StaticMethod()
      prop = OtherClass.StaticMethod();

      // this.prop = OtherClass.InnerClass.StaticMethod()
      prop = OtherClass.InnerClass.StaticMethod();
      // this.prop = OtherClassc.InnerClass.StaticMethodArg("test")
      prop = OtherClass.InnerClass.StaticMethodArg("test");
      // this.prop = OtherClass.InnerClass.StaticMethodArg(propInt.toString())
      prop = OtherClass.InnerClass.StaticMethodArg(propInt.ToString());
    }

  }
}

File: Example/ConvertResults/File/TestMethod.ts
/**
 * メソッドテストクラス
 */
export class TestMethod {
  public static StaticMethod(): string {
    return "ccc";
  }
  /**
   * 定数フィールド
   */
  public ConstField: string = "123";
  private _prop_: string = "";
  /**
   * プロパティ例
   */
  public set prop(value: string) {
    this._prop_ = value;
  }
  /**
   * プロパティ例
   */
  public get prop(): string {
    return this._prop_;
  }
  private _propInt_: number = 50;
  public set propInt(value: number) {
    this._propInt_ = value;
  }
  public get propInt(): number {
    return this._propInt_;
  }
  /**
   * メソッド例
   */
  public Method(): void {
    // this.prop = TestMethod.StaticMethod()
    this.prop = TestMethod.StaticMethod();
    // this.prop = TestLogic.StaticMethod()
    this.prop = TestMethod.StaticMethod();
    // this.prop = TestMethod.InnerClass.StaticMethod()
    this.prop = TestMethod.InnerClass.StaticMethod();
    // this.prop = TestMethod.InnerClass.StaticMethod()
    this.prop = TestMethod.InnerClass.StaticMethod();
    // this.prop = TestMethod.InnerClass.StaticMethodArg("test")
    this.prop = TestMethod.InnerClass.StaticMethodArg("test");
    // this.prop = TestMethod.InnerClass.StaticMethodArg(propInt.toString())
    this.prop = TestMethod.InnerClass.StaticMethodArg(this.propInt.toString());
    // this.prop = OtherClass.StaticMethod()
    this.prop = OtherClass.StaticMethod();
    // this.prop = OtherClass.InnerClass.StaticMethod()
    this.prop = OtherClass.InnerClass.StaticMethod();
    // this.prop = OtherClassc.InnerClass.StaticMethodArg("test")
    this.prop = OtherClass.InnerClass.StaticMethodArg("test");
    // this.prop = OtherClass.InnerClass.StaticMethodArg(propInt.toString())
    this.prop = OtherClass.InnerClass.StaticMethodArg(this.propInt.toString());
  }
}
export namespace TestMethod {
  export class InnerClass {
    public static StaticField: string = "789";
    public static StaticMethod(): string {
      return "bbb";
    }
    public static StaticMethodArg(name: string): string {
      return "hey!" + name;
    }
  }
}
